### ISSUE TITLE
Regression test with Boston Housing data results in 0 accuracy

### ISSUE THREAD

**Issue by rrgiles** [rstudio/keras#30]
library(keras)

cat('Loading data...\n')
bh <- dataset_boston_housing(path = "boston_housing.npz", seed = 113L,test_split = 0.2)
x_train <- bh$train$x
y_train <- bh$train$y
x_test <- bh$test$x
y_test <- bh$test$y

cat(length(x_train), 'train sequences\n')
cat(length(x_test), 'test sequences\n')

model <- keras_model_sequential()

model %>%
  layer_dense(units = 13, input_shape = 13) %>% 
  layer_activation(activation = 'relu') %>%
  layer_dense(units = 1)

# compile model
model %>% compile(
  loss = loss_mean_squared_error,
  optimizer = optimizer_adam(),
  metrics = c('accuracy')
)

# train and evaluate
model %>% fit(
  x_train, 
  y_train,
  batch_size = 5,
  epochs = 100,
  verbose = 1,
  validation_data = list(x_test, y_test)
)
scores <- model %>% evaluate(
  x_test, y_test, verbose = 0
)

cat('Test loss:', scores[[1]], '\n')
cat('Test accuracy:', scores[[2]], '\n')

Output:
Test loss: 71.86488 
Test accuracy: 0

**Comment by topepo** [comment]
Accuracy is probably not what you want for regression problems. It is unlikely that you will get equal numeric values.

**Comment by jjallaire** [comment]
Indeed. See this discussion: https://github.com/fchollet/keras/issues/108.
Accuracy won't mean much for regressions (~ 0 is actually the expected
value here)
